### PR TITLE
Added service account for delegator with DNS permission

### PR DESCRIPTION
**Pull request by savaliyagautam**

### CODE REVIEW

File: delegator.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+resource "google_service_account" "delegator" {
+  account_id   = local.resource_name
+  display_name = "${local.resource_name} service account"
+}
+
+resource "google_project_iam_custom_role" "dns-delegator" {
+  role_id     = replace("dns_delegator_${local.resource_name}", "-", "_")
+  title       = "DNS delegator ${local.resource_name}"
+  description = "DNS delegator for ${local.resource_name}"
+  permissions = ["dns.changes.create", "dns.resourceRecordSets.update", "dns.resourceRecordSets.delete"]
```
Comment (BSick7) [inline delegator.tf:11]: Can we scope these to the dns zone we're creating in this module?

File: outputs.tf [hunk 2]
```diff
@@ -14,6 +14,16 @@ output "zone_id" {
...
+  value       = google_dns_managed_zone.this.name_servers
   description = "list(string) ||| Delegate your managed_zone to these virtual name servers"
 }
+
+output "delegator" {
+  value = {
+    service_account_email = google_service_account.delegator.email
+    key_file              = google_service_account_key.delegator.private_key
+  }
+  description = "object({ service_account_email: string, key_file: string }) ||| "
```
Comment (BSick7) [inline outputs.tf]:
```suggestion
  description = "object({ email: string, key_file: string }) ||| "
```

File: outputs.tf [hunk 3]
```diff
@@ -14,6 +14,16 @@ output "zone_id" {
...
 
 output "nameservers" {
-  value       = google_dns_managed_zone.this.name_servers 
+  value       = google_dns_managed_zone.this.name_servers
   description = "list(string) ||| Delegate your managed_zone to these virtual name servers"
 }
+
+output "delegator" {
+  value = {
+    service_account_email = google_service_account.delegator.email
```
Comment (BSick7) [inline outputs.tf]:
```suggestion
    email = google_service_account.delegator.email
```

File: delegator.tf (hunk 1 above)
Comment (savaliyagautam) [reply delegator.tf:11]: GCP provides IAM conditions to allow permission to specific resources. As per their doc, there isn't support to filter DNS resource https://cloud.google.com/iam/docs/conditions-resource-attributes#resource-name.

File: outputs.tf (hunk 2 above)
Comment (savaliyagautam) [reply outputs.tf]: Taken care

File: outputs.tf (hunk 3 above)
Comment (savaliyagautam) [reply outputs.tf]: Taken care

File: delegator.tf (hunk 1 above)
Comment (BSick7) [reply delegator.tf:11]: Bummer.

**Pull request merged**